File: backend/src/routes/accounts.ts
import { Router, Request, Response } from "express";
import { accountService } from "../services/index.js";

const router = Router();

// GET /api/accounts - List all accounts
router.get("/", (_req: Request, res: Response) => {
  try {
    const accounts = accountService.getAllAccounts();
    res.json(accounts);
  } catch (error) {
    res
      .status(500)
      .json({ error: error instanceof Error ? error.message : "Unknown error" });
  }
});

// GET /api/accounts/:id - Get account by ID
router.get("/:id", (req: Request, res: Response) => {
  try {
    const account = accountService.getAccount(req.params.id);
    if (!account) {
      res.status(404).json({ error: "Account not found" });
      return;
    }
    res.json(account);
  } catch (error) {
    res
      .status(500)
      .json({ error: error instanceof Error ? error.message : "Unknown error" });
  }
});

export default router;

File: backend/app/routers/accounts.py
from fastapi import APIRouter, HTTPException

from app.models import Account
from app.services import db_service

router = APIRouter()


@router.get("", response_model=list[Account])
async def list_accounts():
    """List all accounts."""
    return db_service.get_all_accounts()


@router.get("/{account_id}", response_model=Account)
async def get_account(account_id: str):
    """Get account by ID."""
    account = db_service.get_account(account_id)
    if not account:
        raise HTTPException(status_code=404, detail="Account not found")
    return account
